oauth-common/src/main/java/io/strimzi/kafka/oauth/common/PrincipalExtractor.java

```diff
@@ -88,4 +105,60 @@ public String toString() {
     public boolean isConfigured() {
         return usernameClaim != null || fallbackUsernameClaim != null || fallbackUsernamePrefix != null;
     }
+
+    private static List<String> parseClaimSpec(String spec) {
+        spec = spec == null ? null : spec.trim();
+        if (spec == null || "".equals(spec)) {
+            return null;
+        }
+
+        if (!spec.startsWith("[")) {
+            return Collections.singletonList(spec);
+        }
+
+        LinkedList<String> parsed = new LinkedList<>();
+        int pos = 0;
+        int epos = 0;
+        while (pos != -1) {
+            pos += 1;
+            epos = spec.indexOf("]", pos);
+            if (epos == -1) {
+                throw new IllegalArgumentException("Failed to parse username claim spec: '" + spec + "' (Missing ']')");
+            }
+            parsed.add(removeQuotesAndSpaces(spec.substring(pos, epos)));
+            pos = epos + 1;
+            epos = skipSpaces(spec, pos);
+            if (epos >= spec.length()) {
+                return parsed;
+            }
+            if (spec.charAt(epos) != '.') {
+                throw new IllegalArgumentException("Failed to parse usename claim spec: '" + spec + "' (Missing '.' at position: " + epos + ")");
+            }
+            pos = spec.indexOf("[", epos + 1);
```

Comment:
How does this handle the input `[foo].bar[baz]`?

Reply:
This should be illegal. It should throw an exception. I'll fix it. Once you start your claim spec with '[' the characters between ']' and '[' should be limited to arbitrary number of spaces and a single '.'.

Reply:
Fixed.